mindmap/static/js/src/mindmap.js

```diff
@@ -19,4 +21,17 @@ function MindMapXBlock(runtime, element) {
 
     var jm = new jsMind(options);
     jm.show(mind);
+
+    $(element).find('.save-button').click(function () {
+        const mindMapData = jm.get_data('node_array');
+        const jsonMindMapData = JSON.stringify(mindMapData);
+        var handlerUrl = runtime.handlerUrl(element, 'upload_file');
+        var data = { mind_map: jsonMindMapData };
+
```

Comment:
Something that I have found new is that we can [notify](https://github.com/openedx/edx-platform/blob/b94b0a8e40672cff7a5ff6c3a69565b0635f273f/cms/static/cms/js/spec/xblock/cms.runtime.v1_spec.js#L25) the user with alerts :dancers: 

```
const { notify } = new window.StudioRuntime.v1();

// At the beginnig
notify("save", {
  state: "start",
  message: "Saving...",
});

// At the end
notify("save", {
  state: "end",
});

// For errors:

notify("error", {
  title: "title",
  message: "message",
});

```

So, you can use like this:

```
const { notify } = new window.StudioRuntime.v1();

notify("save", {
  state: "start",
  message: "Saving...",
});

$.post(handlerUrl, JSON.stringify(data))
  .done(function () {
    window.location.reload(false);
    notify("save", {
      state: "end",
    });
  })
  .fail(function () {
    notify("error", {
      title: "Error",
      message: "Error saving mind map",
    });
    console.log("Error saving mind map");
  });

```

Reply:
This is not working, what should it show? a popup?

Reply:
Please delete notifications alert and add a comment with a TOD

`// TODO add notifications`

Reply:
If this shows a message saying `Saving...` then it's a great idea! I was about to comment about adding some kind of feedback so the user knows their file is being saved

Reply:
Also, we should disable the button when uploading the file

Reply:
We can consider these changes in another PR